### ISSUE TITLE
Weird super slow-revising edge case

### ISSUE THREAD

**Issue by marius311** [timholy/Revise.jl#596]
I hesitated almost to file this but I spent some time finding a MWE so I figure why not. It has a fine workaround so please feel free to just close. Basically, the following module is extremely slow to revise, about 7 seconds every time I change `x`. 

```julia
module Foo

x = 1

using Pkg, Zygote, StaticArrays

versionof(pkg::Module) = Pkg.dependencies()[Base.PkgId(pkg).uuid].version
if versionof(Zygote) > v"0.4.15"
    Zygote._zero(xs::StaticArray, T) = SizedArray{Tuple{size(xs)...},Union{T,Nothing}}(map(_->nothing, xs))
end

end
```

Not sure what's happening, the `versionof` function runs in <<1 second. The workaround is to make that a `@static if` (maybe thats a helpful suggestion in other cases).

**Comment by timholy** [comment]
Confirmed. If you ever want more insight about what's going on, putting `@show mod ex` right before https://github.com/timholy/Revise.jl/blob/a7b51cb32e71b46153c01b1f3c1c7d587d3932f5/src/lowered.jl#L180

is usually pretty informative. I'll take a look, though.

**Comment by timholy** [comment]
So, because `versionof` runs in the interpreter, the first revision will still be slow, but the problem was Revise was failing to take credit for previous work it had done. Fixed in #603.

**Comment by timholy** [comment]
BTW, this may also fix bugs (never reported) in the handling of `@require` blocks, so thanks for reporting this!
